packages/runtime/container-runtime/src/opLifecycle/opDecompressor.ts

```diff
@@ -44,12 +45,18 @@ export class OpDecompressor {
             const asObj = JSON.parse(intoString);
             this.rootMessageContents = asObj;
 
-            return { ...message, contents: this.rootMessageContents[this.processedCount++] };
+            return {
+                message: { ...message, contents: this.rootMessageContents[this.processedCount++] },
+                state: "Accepted",
+            };
         }
 
         if (this.rootMessageContents !== undefined && message.metadata?.batch === undefined && this.activeBatch) {
             // Continuation of compressed batch
-            return { ...message, contents: this.rootMessageContents[this.processedCount++] };
+            return {
```

Comment:
It would be great to leverage defensive programming here and assert that op content is empty.
It also would be great to assert that this op comes from the same clientId, i.e. batch is not interleaved.
It's hard to say if it will be possible with your changes alone (do we run this code before we received all chunks? If so, interleaving will be possible), but it's definitely something that will happen once we start implementing sequence virtualization and removal of batch concept from protocol. We will need to fix all the code that relies on that, but it might be hard to find all such code, so code being defensive will help next developer to realize all such dependencies.

Reply:
>  this op comes from the same clientId, i.e. batch is not interleaved

This should not concern the decompressor. Batch interleaving is handled by the schedule manager and it's on a higher level of abstraction IMO. We can assert on the actual content (like op content being empty when expected) but the decompressor should be oblivious to concepts like clientId.